### PR TITLE
[ExoBundle] Order Steps and Questions

### LINKED ISSUE #113: ExoBundle : Gestion questions/steps dans le player
_From @charlotte42 on March 31, 2016 8:42_

Pouvoir changer l'ordre des questions dans une étape et les déplacer d'une step à l'autre
Importer des questions directement dans une step

_Copied from original issue: claroline/ClaroSupport#31_

### PR DESCRIPTION
**Pull request by Elorfin**
Add the drag n drop feature to order Steps and Questions.

It also includes little bug fixes and refactoring.

Will fix #113.

### CODE REVIEW

File: plugin/exo/Controller/Api/StepController.php [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+     * @param Request  $request
+     *
+     * @return JsonResponse
+     */
+    public function reorderAction(Exercise $exercise, Request $request)
+    {
+        $this->assertHasPermission('ADMINISTRATE', $exercise);
+
+        $dataRaw = $request->getContent();
+        if (!empty($dataRaw)) {
```
Comment (stefk) [inline plugin/exo/Controller/Api/StepController.php]: An empty request shouldn't be valid (204) IMO.

File: plugin/exo/Controller/Api/StepController.php [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+     * @param Request  $request
+     *
+     * @return JsonResponse
+     */
+    public function reorderQuestionsAction(Exercise $exercise, Step $step, Request $request)
+    {
+        $this->assertHasPermission('ADMINISTRATE', $exercise);
+
+        $dataRaw = $request->getContent();
+        if (!empty($dataRaw)) {
```
Comment (stefk) [inline plugin/exo/Controller/Api/StepController.php]: Same.

**Comment by stefk** [comment]
(LGTM)

**Comment by LaurentGruber** [comment]
LGTM

**Comment by Elorfin** [comment]
@LaurentGruber : too fast, I would have liked to reduce the number of commits ;).

**Comment by LaurentGruber** [comment]
@Elorfin sorry :s

**Pull request merged**